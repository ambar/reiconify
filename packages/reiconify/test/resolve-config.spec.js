const {resolveConfig} = require('..')

describe('resolveConfig', () => {
  const cwd = process.cwd()

  beforeEach(() => {
    process.chdir(__dirname)
  })

  afterAll(() => {
    process.chdir(cwd)
  })

  it('gets default config', async () => {
    expect(await resolveConfig()).toMatchSnapshot()
  })

  it('overwrites default config', async () => {
    process.chdir('fixtures/resolve-config/')
    expect(await resolveConfig()).toMatchSnapshot()
  })
})
