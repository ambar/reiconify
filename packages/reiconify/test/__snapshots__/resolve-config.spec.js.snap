// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolveConfig gets default config 1`] = `
{
  "baseClassName": "",
  "baseDefaultProps": {
    "fill": "currentColor",
  },
  "baseName": undefined,
  "baseTemplate": [Function],
  "camelCaseProps": true,
  "defaultProps": {},
  "filenameTemplate": [Function],
  "indexTemplate": [Function],
  "name": "Icon",
  "svgoPlugins": [],
  "template": [Function],
}
`;

exports[`resolveConfig overwrites default config 1`] = `
{
  "baseClassName": "MyIcon",
  "baseDefaultProps": {
    "fill": "currentColor",
  },
  "baseName": undefined,
  "baseTemplate": [Function],
  "camelCaseProps": true,
  "defaultProps": {},
  "filenameTemplate": [Function],
  "indexTemplate": [Function],
  "name": "Icon",
  "svgoPlugins": [
    {
      "removeAttrs": {
        "attrs": "fill",
      },
    },
  ],
  "template": [Function],
}
`;
